packages/atomic/src/components/result-lists/result-list-common.tsx

```diff
@@ -235,38 +237,46 @@ export class ResultListCommon {
       return;
     }
 
+    if (resultListState.firstSearchExecuted && !resultListState.hasResults) {
+      return;
+    }
+
+    const displayPlaceholders = !isAppLoaded(this.bindings.store);
```

Comment:
I don't know if you explained this before, but what is the difference between `isAppLoaded(this.bindings.store);` and `resultListState.firstSearchExecuted`?

Reply:
It used to look for these variables but it wasn't enough, the resultListState is the status of headless related to the search call. We need more than that to know that the results in Atomic have truly loaded:

1. The Stencil web component defined in the HTML are mounted, that mean's each file is being loaded separately for each component
2. initialize() method is called on the parent interface and on each of the registered components in the page
3. the result list loads it's result template children result template
4. executeFirstSearch() method is called on the parent interface, `resultListState.isLoading` is now `true`
5. the search fulfills and the results are loaded in the state `resultListState.isLoading` is now `false` & `resultListState.firstSearchExecuted` is true. This is the start of the "gap" where placeholders were hidden but results not yet visible
6. the result list renders the proper results using one of the result templates
7. now that the result template components are in the HTML, the related files are being loaded in order to render each components
8. once the components have loaded & rendered, the `isAppLoaded` should return `true`, this is the end of the visual "gap"

Reply:
Maybe i can make the following more clear ? https://github.com/coveo/ui-kit/pull/1984/files#diff-6fb7deead3ab9ab5f889dad6759d737983382cc2594ccf00f31a16d99099af82R35-R40